### PR TITLE
NIFI-10207: ensure allowableValues matches SUPPORTED_STATE in Process…

### PR DESCRIPTION
**Pull request by ottobackwards**
…orRunStatusEntity

# Summary

[NIFI-10207](https://issues.apache.org/jira/browse/NIFI-10207)

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [x] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [x] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [x] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [x] Pull Request based on current revision of the `main` branch
- [x] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [x] Build completed using `mvn clean install -P contrib-check`
  - [x] JDK 8
  - [ ] JDK 11
  - [ ] JDK 17

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

**Comment by ottobackwards** [comment]
verified after build 
```json
    "ProcessorRunStatusEntity" : {
      "type" : "object",
      "properties" : {
        "revision" : {
          "description" : "The revision for this request/response. The revision is required for any mutable flow requests and is included in all responses.",
          "$ref" : "#/definitions/RevisionDTO"
        },
        "state" : {
          "type" : "string",
          "description" : "The run status of the Processor.",
          "enum" : [ "RUNNING", "STOPPED", "DISABLED", "RUN_ONCE" ]
        },
        "disconnectedNodeAcknowledged" : {
          "type" : "boolean",
          "description" : "Acknowledges that this node is disconnected to allow for mutable requests to proceed."
        }
      },
      "xml" : {
        "name" : "entity"
      }
    }
```

**Comment by ottobackwards** [comment]
@tpalfy

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/ProcessorRunStatusEntity.java [hunk 1]
```diff
@@ -26,6 +26,9 @@
 @XmlType(name = "processorRunStatus")
 public class ProcessorRunStatusEntity extends ComponentRunStatusEntity {
 
+    // SUPPORTED_STATE *MUST* match the allowableValues below
+    // if it doesn't, OpenAPI/Swagger generation will generate
+    // correct clients, and they will have errors when deserializing
```
Comment (tpalfy) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/ProcessorRunStatusEntity.java:31]: I think a simpler but completely specific reminder comment would be better.

All we really want is to make sure when the constant changes, the annotation is updated as well. The rest is technical detail, one that could be written to almost all similar places.

I'd do something like this:

```suggestion
    // Must match the @ApiModelProperty.allowableValues above getState()
```

**Comment by tpalfy** [comment]
LGTM
Thanks for your work @ottobackwards!

Pushed to main.

**Pull request closed without merging**